src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -1,395 +1,372 @@
+        {
+            if (!this.NuGetPackages.Any())
+            {
+                return;
+            }
+
+            Debug.Assert(this.NuGetPackages.Count == this.NuGetPackages.Distinct().Count(), "Duplicate NuGet packages specified");
+
+            var managedProjects = this.projectSystemHelper.GetSolutionManagedProjects().ToArray();
+            if (!managedProjects.Any())
+            {
+                Debug.Fail("Not expected to be called when there are no managed projects");
+                return;
+            }
+
+            DeterminateStepProgressNotifier progressNotifier = new DeterminateStepProgressNotifier(notificationEvents, managedProjects.Length * this.NuGetPackages.Count);
+            foreach (var project in managedProjects)
+            {
+                foreach (var packageInfo in this.NuGetPackages)
+                {
+                    if (token.IsCancellationRequested)
+                    {
+                        break;
+                    }
+
+                    string message = string.Format(CultureInfo.CurrentCulture, Strings.EnsuringNugetPackagesProgressMessage, packageInfo.Id, project.Name);
+                    progressNotifier.NotifyCurrentProgress(message);
+
+                    // TODO: SVS-33 (https://jira.sonarsource.com/browse/SVS-33) Trigger a Team Explorer warning notification to investigate the partial binding in the output window.
+                    this.AllNuGetPackagesInstalled &= NuGetHelper.TryInstallPackage(this.owner.ServiceProvider, project, packageInfo.Id, packageInfo.Version);
+
+                    progressNotifier.NotifyIncrementedProgress(string.Empty);
+                }
+            }
+        }
+
+        internal /*for testing purposes*/ void EmitBindingCompleteMessage(IProgressStepExecutionEvents notifications)
+        {
+            var message = this.AllNuGetPackagesInstalled
+                ? Strings.FinishedSolutionBindingWorkflowSuccessful
+                : Strings.FinishedSolutionBindingWorkflowNotAllPackagesInstalled;
+            notifications.ProgressChanged(message, double.NaN);
+        }
+        #endregion
+
+        #region Helpers
+        private RuleSetGroup LanguageToGroup(string language)
+        {
+            RuleSetGroup group;
+            if (!this.LanguageToGroupMapping.TryGetValue(language, out group))
+            {
+                Debug.Fail("Unsupported language: " + language);
+                throw new InvalidOperationException();
+            }
+            return group;
+        }
+
+        private void AbortWorkflow(IProgressController controller, CancellationToken token)
+        {
+            bool aborted = controller.TryAbort();
+            Debug.Assert(aborted || token.IsCancellationRequested, "Failed to abort the workflow");
+        }
+        #endregion
+
+    }
+}
```

Comment:
Minor: Terminating new lines are preferred in the git-world. Many tools (including GitHub diff) will show a warning about the lack of newline.